lib/puma/server.rb

```diff
@@ -478,6 +481,19 @@ def process_client(client, buffer)
       end
     end
 
+    # Compute maximum time to wait for another request arriving on
+    # a client when keep-alive is enabled.
+    # If the pool we created has a maximum of one worker thread we
+    # can't afford to let it wait for the next request while other
+    # work might be pending.
+    def fast_ka_check
```

Comment:
I'd document that this method can return `nil`. It's important because `nil`is a valid value for `Client#reset`.